File: server/mc/1.21.1-neoforge/src/main/java/calebxzhou/rdi/mc/server/WsHandler211.java
package calebxzhou.rdi.mc.server;

import calebxzhou.rdi.mc.common.WebSocketClient;
import calebxzhou.rdi.mc.common.WsMessage;
import calebxzhou.rdi.mc.common.WsMessageHandler;
import calebxzhou.rdi.mc.common.RGlobalPlayerList;
import calebxzhou.rdi.mc.rcmd.chat.RChatMessage;
import calebxzhou.rdi.mc.server.network.RServerNetwork;
import com.google.gson.JsonElement;
import net.minecraft.network.chat.Component;
import net.minecraft.server.dedicated.DedicatedServer;

/**
 * calebxzhou @ 2026-01-12 19:54
 */
public class WsHandler211 implements WsMessageHandler {
    private final DedicatedServer server;

    public WsHandler211(DedicatedServer server) {
        this.server = server;
    }

    @Override
    public void onMessage(WsMessage<JsonElement> msg) {
        switch (msg.getChannel()){
            case Command -> {
                var cmd = msg.getData().getAsString();
                var resp = server.runCommand(cmd);
                WebSocketClient.sendMessage(msg.getId(), WsMessage.Channel.Response, resp);
            }
            case Chat -> {
                var chatMessage = WebSocketClient.fromJson(msg.getData(), RChatMessage.class);
                server.getPlayerList().broadcastSystemMessage(Component.literal("[公共] " + chatMessage.playerName + ": " + chatMessage.content), false);
            }
            case PlayerList -> {
                var playerList = WebSocketClient.fromJson(msg.getData(), RGlobalPlayerList.class);
                RServerNetwork.sendToAll(server, playerList);
            }
            default -> {}
        }
    }
}

File: server/mc/1.21.1-neoforge/src/main/kotlin/calebxzhou/rdi/mc/server/WsHandler211.kt
package calebxzhou.rdi.mc.server

import calebxzhou.rdi.mc.common.RGlobalPlayerList
import calebxzhou.rdi.mc.common.WebSocketClient
import calebxzhou.rdi.mc.common.WsMessage
import calebxzhou.rdi.mc.common.WsMessageHandler
import calebxzhou.rdi.mc.rcmd.chat.RChatMessage
import calebxzhou.rdi.mc.server.network.RServerNetwork
import com.google.gson.JsonElement
import net.minecraft.network.chat.Component
import net.minecraft.server.dedicated.DedicatedServer

/**
 * calebxzhou @ 2026-01-12 19:54
 */
class WsHandler211(private val server: DedicatedServer) : WsMessageHandler {
    override fun onMessage(msg: WsMessage<JsonElement>) {
        when (msg.getChannel()) {
            WsMessage.Channel.Command -> {
                val cmd = msg.getData()!!.getAsString()
                val resp = server.runCommand(cmd)
                WebSocketClient.sendMessage<String?>(msg.getId(), WsMessage.Channel.Response, resp)
            }

            WsMessage.Channel.Chat -> {
                val chatMessage = WebSocketClient.fromJson<RChatMessage>(msg.getData(), RChatMessage::class.java)
                server.getPlayerList().broadcastSystemMessage(
                    Component.literal("[公共] " + chatMessage.playerName + ": " + chatMessage.content),
                    false
                )
            }

            WsMessage.Channel.PlayerList -> {
                val playerList =
                    WebSocketClient.fromJson<RGlobalPlayerList?>(msg.getData(), RGlobalPlayerList::class.java)
                RServerNetwork.sendToAll(server, playerList)
            }

            else -> {}
        }
    }
}